databasetest/databasetest.go

```diff
@@ -0,0 +1,49 @@
+// Copyright 2023 The Authors (see AUTHORS file)
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package databasetest provides an ephemeral database server for integration testing.
+// It's designed to be used in code that needs to work inside and outside google.
+package databasetest
+
+import "io"
+
+// ConnInfo specifies how to connect to the created container.
+type ConnInfo struct {
+	Hostname string
+	// PortMapper maps from container port to host port. Do not use after container is closed.
+	PortMapper func(string) string
+}
+
+type Driver interface {
+	ImageRepository() string // Repository for docker image (ex: mysql)
```

Comment:
I'm not sure, but I think the comments need to appear above the field they describe. Otherwise they won't show up properly in godoc.

Reply:
https://bradleyjkemp.dev/godoc-playground/

From what I can tell playing with that and the `go doc` command, godoc doesn't parse interface comments or struct field comments. It just puts the entire interface/struct in a codeblock comments and all, so its purely a style/readability concern.